metrics/cloudwatch/cloudwatch.go

```diff
@@ -44,6 +46,15 @@ func New(namespace string, svc cloudwatchiface.CloudWatchAPI, logger log.Logger)
 	}
 }
 
+// SetConcurrency overrides the default number (10) of concurrent requests sent to CloudWatch.
+// CloudWatch allows a maximum of 20 metrics to be sent per request, so when Send is called,
+// we partition the metrics and concurrently call their API. This parameter sets maximum number
+// of concurrent requests.
+func (cw *CloudWatch) SetConcurrency(numConcurrentRequests int) *CloudWatch {
+	cw.numConcurrentRequests = numConcurrentRequests
+	return cw
+}
```

Comment:
No, that's not the way. If I were you I'd just make it a non-optional parameter, and document a reasonable default in the function comment.

```go
// New returns a blah blah.
// The client will emit at most numConcurrent simultaneous requests to Amazon.
// A good default value is 10.
func New(namespace string, svc CloudWatchAPI, numConcurrent int, logger log.Logger) {
    ...
}
```